manifests/root_user.pp

```diff
@@ -28,6 +32,16 @@
   }
 
   if $manage_user {
+    case $password {
+      undef:                { $_password = undef }
+      /^\$6\$/:             { $_password = Sensitive($password) }
+      /^\$[125]{1}[ay]?\$/: { fail('Error: You cannot use MD5, Blowfish, or SHA256 hashing algorithms for the user password. Please hash with SHA512.') }
```

Comment:
These are actually valid since we don't actually *know* what's configured on the remote system.

According to `shadow(5)` and `crypt(5)`, the valid values are:

* Something that starts with `!` (locked)
* `$<something>$<more stuff>`

In theory, the data type could be as simple as `Pattern['(!|\$.+\$).+]` but I haven't tested that.

Reply:
Yeah, they're valid, but not secure. I guess I can check the configuration for FIPS or create some security/STIG control setting that allows enforcement of the SHA-512 value and fails on the rest of the options for enforcement.